src/main/scala/com/fortysevendeg/android/sqlite/resultset/SQLDroidResultSet.scala

```diff
@@ -0,0 +1,688 @@
+package com.fortysevendeg.android.sqlite.resultset
+
+import java.io._
+import java.math.BigDecimal
+import java.net.URL
+import java.sql._
+import java.text.SimpleDateFormat
+import java.util
+import java.util.Calendar
+
+import android.database.Cursor
+import com.fortysevendeg.android.sqlite._
+
+import scala.util.{Failure, Success, Try}
+import scala.{Array => SArray}
+
+class SQLDroidResultSet(
+  cursor: Cursor,
+  log: LogWrapper = new AndroidLogWrapper())
+  extends ResultSet
+  with NonEditableResultSet
+  with WrapperNotSupported {
+
+  private[this] var lastColumnRead: Option[Int] = None
```

Comment:
Don't use `vars`. I would recommend you study two alternatives:

- StateT Monad: http://eed3si9n.com/learning-scalaz/State.html
- FSM: http://stackoverflow.com/questions/18226863/general-finite-state-machine-transducer-in-scala